src/app/auth/auth.js

```diff
@@ -0,0 +1,152 @@
+angular.module('prx.auth', ['prx.ui.nav'])
+.config(function (PRXDrawerProvider) {
+  PRXDrawerProvider.register({
+    name: 'Sign In',
+    weight: PRXDrawerProvider.BOTTOM,
+    href: 'http://www.prx.org/sessions/new',
+    nav: true,
+    template: '<prx-auth-badge></prx-auth-badge>',
+    dropdownTemplate: '<prx-auth-window></prx-auth-window>'
+  });
+}).service('PrxAuth', function (ngHal, $q, $window) {
+  var PrxAuth = this,
+      deferred = $q.defer(),
+      currentUser = {loggedIn: false};
+  this.$$loginPromise = deferred.promise;
+
+  this.$checkLoggedIn = function () {
+    var iframe = angular.element('<iframe>'),
+        $body = angular.element($window.document.body),
+        uri = [FEAT.ID_SERVER + '/authorize?client_id=' + FEAT.ID_CLIENT_KEY],
+        nonce = [];
+    for (var i=0; i<8; i++) {
+      nonce.push(randomInt(0, 15).toString(16));
+    }
+    nonce = nonce.join('');
+    uri.push("&nonce=", nonce, "&response_type=token&prompt=none");
```

Comment:
`prompt=none` means that we are guaranteed to get a success or failure callback from the auth server immediately without needing any user interaction. Or, at least, if user interaction is required, we will consider that an error state and call back with that.